### ISSUE TITLE
Increased the number of apparently "ramdonly" missed blocks due to collision since HF20.

### ISSUE THREAD

**Issue by pgarciagon** [steemit/steem#3087]
Since the deployment of v0.20.5 several top witness have been reporting unexpected missed blocks with automatic recovery after a while.  A close follow-up of possible future misses could be perform in this issue.

On my last misses @cervantes was "outforked" on:

```
[[33m2867799ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 019abd6e36de2011887188c955a58bb71b5f3d32, 26918254^M^[[0m
[[33m2867799ms fork_database.cpp:44          push_block           ] Head: 26918252, 019abd6c61c765de1a73c91764f428ea8dbbad11^M^[[0m
^M^[[31m2867838ms p2p_plugin.cpp:229            handle_block         ] Error when pushing block:
4080000 unlinkable_block_exception: unlinkable block
block does not link to known chain
    {}
    fork_database.cpp:64 _push_block

    {}
    database.cpp:782 _push_block

```

and recovered after while with:

````
^M^[[0m2923313ms database.cpp:690              _maybe_warn_multiple ] Encountered block num collision at block 26918254 due to a fork, witnesses are: [["cervantes","2018-10-18T14:48:03"],["timcliff","2018-10-18T14:47:48"]]^M^[[0m
^M^[[33m2924130ms database.cpp:716              _push_block          ] Switching to fork: 019abd6f0e99f443c210213f6874ae109e6a875f^M^[[0m
^M^[[0m2924136ms database.cpp:726              _push_block          ] pushing blocks from fork 26918254 019abd6e36de2011887188c955a58bb71b5f3d32^M^[[0m
^M^[[0m2924140ms database.cpp:726              _push_block          ] pushing blocks from fork 26918255 019abd6f0e99f443c210213f6874ae109e6a875f^M^[[0m
^M^[[33m2925596ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 019abd7f99d3fb55c6b4bef8d6d3992702fe3505, 26918271^M^[[0m
^M^[[33m2925596ms fork_database.cpp:44          push_block           ] Head: 26918258, 019abd722d8592b1dee646eb07efc6deb37a54eb^M^[[0m
^M^[[31m2925665ms p2p_plugin.cpp:229            handle_block         ] Error when pushing block:
4080000 unlinkable_block_exception: unlinkable block
block does not link to known chain
    {}
    fork_database.cpp:64 _push_block

    {}
    database.cpp:782 _push_block

```

**Comment by Gandalf-the-Grey** [comment]
Works for me:
I've never experienced this during normal operation (i.e. after initial sync to the head block) on primary or backups (checked on 4 nodes, including my very busy public seed node, complete logs since `v0.20.5` was introduced)

```
968119ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 019993e167582c7e518cc62c4ff215bbc36e194b, 26842081
968119ms fork_database.cpp:44          push_block           ] Head: 26840245, 01998cb5de9c77729fc6212d77ac763b84e0f6e8
```

```
1435979ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 019276571435001145865a90f6ee0953c29c9b27, 26375767
1435979ms fork_database.cpp:44          push_block           ] Head: 26367102, 0192547edecf969fb93fa153dac2deabec1db54a
```

```
870156ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01925e321bcda178d011c3a985d5793c735c1d09, 26369586
870156ms fork_database.cpp:44          push_block           ] Head: 26363833, 019247b953769aaaeb86a4ca954d40ed705cea4b
```

```
1515258ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01925a5a03d809741bbf4722bd2eb83bc2945948, 26368602
1515258ms fork_database.cpp:44          push_block           ] Head: 26363769, 01924779df48796b2f22adfc8880660e30cf5d4d
```

Update: I have one node that encountered three events but those can be explained by not the best networking out there.
```
3473859ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 0199892c8019a917d59953ba23d7cf179f853cec, 26839340
3473859ms fork_database.cpp:44          push_block           ] Head: 26839338, 0199892a22f6cbd04110c867318acd181223f968
3477181ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 0199892db1b263ee32683feefdb06565eaabc584, 26839341
3477181ms fork_database.cpp:44          push_block           ] Head: 26839339, 0199892bf4b20399e955f5f328e8f3512e52a59b
3479648ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 0199892e79abf6bb925e6942d06fca0d6cdc6e4d, 26839342
3479648ms fork_database.cpp:44          push_block           ] Head: 26839339, 0199892bf4b20399e955f5f328e8f3512e52a59b
3482674ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 0199892f62c2acb1aa3a8a543381b6ba532f8faf, 26839343
3482674ms fork_database.cpp:44          push_block           ] Head: 26839339, 0199892bf4b20399e955f5f328e8f3512e52a59b
3487608ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 0199893054192ed511d0538ea981988489fd7f1d, 26839344
3487608ms fork_database.cpp:44          push_block           ] Head: 26839339, 0199892bf4b20399e955f5f328e8f3512e52a59b
3488668ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 019989315973fffc2dd55f80a5444acde072dc8c, 26839345
3488669ms fork_database.cpp:44          push_block           ] Head: 26839339, 0199892bf4b20399e955f5f328e8f3512e52a59b
```

```
1844713ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb00977f19b9f7ae0902ba957dc5fc2af82, 26839984
1844713ms fork_database.cpp:44          push_block           ] Head: 26839982, 01998bae73e23aac697d375fa93fcaedd8d0342d
1847688ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb1cccdcbc9eaea4264222feefb308be320, 26839985
1847688ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1850675ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb2b5817eba8392ca997dfe8b5c9c7591b3, 26839986
1850675ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1854105ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb3ee9bea02845d33c76dae8e652ae9a813, 26839987
1854105ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1856780ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb40d59d127cbc0bfcb22ca0e92ee4a260e, 26839988
1856780ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1860196ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb5bb5583097b8ae1a7fafc8db3eeaefd60, 26839989
1860196ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1862657ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb67636fb2bc0c9fec13fc36be3f2c14849, 26839990
1862657ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1865974ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb74b3f2a001e45f02df7aa7de7bd539ea7, 26839991
1865974ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1868650ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb804cfe4c7b24e445e8b3a7781f9a36f9d, 26839992
1868650ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1872191ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bb9849a211b4c408b0a976e870d2f69aa2a, 26839993
1872191ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1874781ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bbae811eb7b55fc57ea10f4124690b9902f, 26839994
1874781ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1877679ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bbbfc6332a47c12aef095142179040ec84e, 26839995
1877679ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1881072ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bbc69c991dc7edcc85514e9a5a783c5e4f2, 26839996
1881073ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
1883680ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998bbd238fd93e0bc6fddfa92db9838195e731, 26839997
1883680ms fork_database.cpp:44          push_block           ] Head: 26839983, 01998bafd2d96796638f9f570f612707d2386715
```

```
140740ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e21d794bcb754f2834e7dc6d998957512b6, 26840609
140740ms fork_database.cpp:44          push_block           ] Head: 26840607, 01998e1ff02c77b486ff3e8bd111c7c04ee91063
143704ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2234c430d1f286653d4f1df5e33f17d26b, 26840610
143704ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
147112ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e233060d0826e085456b5a8d4e073d09e59, 26840611
147112ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
149903ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2478668799880eebe3f90df1d6e81c078b, 26840612
149903ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
152665ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2540ad990f45f3bd194bb503f972081545, 26840613
152665ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
155927ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e26afbd5d85574a20fa4c34eeb9459ca21c, 26840614
155927ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
158665ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2757c32b93d281a7a5545340730585a05f, 26840615
158665ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
162384ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2804ec0c50f697c753d0c642801d4e8fcc, 26840616
162384ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
164965ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e29f1181c6859b96192ebd75e3f3bb0bf21, 26840617
164965ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
167664ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2a05509870a5fbb173bdf623255df7f45e, 26840618
167664ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
170681ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2bbcd20f88db6082e243252ef6dae6dbbe, 26840619
170681ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
173673ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2c7df5173bb383522b6c6fe2090347cb2a, 26840620
173673ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
177204ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2d121e0f9471c64764ab39e542c3b82a80, 26840621
177204ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
179712ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2e46a7c8550784c7179576d98562bd258e, 26840622
179712ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
182696ms fork_database.cpp:43          push_block           ] Pushing block to fork database that failed to link: 01998e2ff4b724b96bea2237582d14317a329b59, 26840623
182696ms fork_database.cpp:44          push_block           ] Head: 26840608, 01998e20914df29c5f1549914ef128570ca4209f
```

TLDR: Nothing unusual here.

**Comment by liondani** [comment]
I think also it is just latency related since the code now is obviously more strict on accepting blocks "in time"

**Comment by youkaicountry** [comment]
Looks like the issue is due to latency. Please reopen if it turns out otherwise.
